scipy/linalg/tests/test_lapack.py

```diff
@@ -406,6 +406,19 @@ def test_gelsy(self):
                             rtol=25*np.finfo(dtype).eps)
 
 
+@pytest.mark.parametrize('dtype', DTYPES)
+def test_geqrf_lwork(dtype):
+    geqrf, geqrf_lwork = get_lapack_funcs(
+        ('geqrf', 'geqrf_lwork'),
+        dtype=dtype,
+    )
+    m, n = 4, 3
```

Comment:
This is a quite fast query hence don't be shy to test with big numbers (less than 2**16 and maybe just to see what happens larger than 2**16).

Reply:
I gave it a run and it's failing for the complex128 with a memory error. The error comes from the call to ``np.ones`` - so it's not necessarily that the wrapper is failing  - but my hardware's ability to store the larger array.

Reply:
You don't need to call the actual `geqrf`, we are calling it anyways with this wrapper. As long as info=0 or nonzero we are fine.

The time is spent in creating the ones matrix actually.

Reply:
Ok. I called ``geqrf(...,-1)`` first to have a value to check if it is correct. If I understand you correctly, we don't need to do that. I've got something on tonight - I'll push the changes once I get home.